### PR TITLE
refactor(source-mixpanel): migrate to CDK v3

### PR DESCRIPTION
**Pull request by lazebnyi**
## What

migrate to CDK v3; 
Fix https://github.com/airbytehq/airbyte-internal-issues/issues/8387

## How

migrate:
- `def backoff_time` -> `def get_backoff_strategy`
- `def should_retry` -> `def get_error_handler`

## Can this PR be safely reverted and rolled back?
- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (brianjlai) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:209]: for any mypy ignore w/ misc, we should add a comment for why we're ignoring this since it's not obvious why we couldn't fix this

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (brianjlai) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:213]: These lines presumably fail because `stream_slice` is optional. Rather than ignore this, we should be defining what the behavior is supposed to be if the input is `None`. For GH is this actually a possibility?

So two asks are:

- We should define the correct behavior (should `to_date` still exist in the mapping if `end_date` does not exist in `stream_slice`)?
-  Can we get rid of the `ignore` if we were to add an appropriate `None` check so that when we call `stream_slice["end_date"] we know this is always a mapping

**Review by brianjlai — COMMENTED** [review]
Added some initial comments, will continue reviewing some of the custom components and error handlers, but wanted to first discuss the large amount of `ignore` mypy statements.

I think that at best, there are some places where we can fix the actual mypy issues instead of ignoring them (see one of my comments below), and at worst, we need to include more comments on the annotations to explain why we have to ignore the type check

File: airbyte-integrations/connectors/source-mixpanel/pyproject.toml
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/pyproject.toml]: updated

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/utils.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/utils.py]: updated

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py]: updated

**Comment by lazebnyi** [comment]
@brianjlai We had a discussion with @girarda and decided that, since mypy checks are not required for connectors, we can skip fixing those issues and focus only on changes related to error handling.

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/errors_handlers/base_errors_handler.py
Comment (brianjlai) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/errors_handlers/base_errors_handler.py:14]: Instead of redefining a new `MixpanelStreamErrorHandler`, can we instead use the existing `HttpStatusErrorHandler` and pass in a custom `error_mapping` into `__init__` when we instatiate the handler in `base.py`?

If that is not possible to do, we should add a comment to this and the second error handler about why these custom implementation were needed. When we can avoid creating child classes of the existing components the better

**Review by brianjlai — COMMENTED** [review]
I have some comments about how we're structuring the custom components, but I think this is very close

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py:78]: I comment this just to simplify e2e tests with rate limits

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:112]: same as above

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (artem1205) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:116]: max_retries should be passed to error_handler as parameter

Comment (artem1205) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:116]: Yes, but it is `5`, while here it is set to `3`.
https://github.com/airbytehq/airbyte/blob/f09a615affc92dfa9b1ebb6c5355f945816e2036/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/http_status_error_handler.py#L16-L23

 Do we want to use default value?

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py]: reverted

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py]: reverted

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/errors_handlers/base_errors_handler.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/errors_handlers/base_errors_handler.py:14]: Added docs to error handler classes

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:116]: Yes, there's no necessary reason to use 3 retries instead of the default 5 retries.

**Review by artem1205 — APPROVED** [review]

**Review by brianjlai — APPROVED** [review]
looks good to me thanks for making the changes asked. But since regression tests were last run 2 weeks ago, can you kick off one more regression test run before merging?

**Review by pnilan — APPROVED** [review]

**Comment by pnilan** [comment]
@lazebnyi Hey where does this PR stand? Looks like it should be ready to merge after resolving conflicts?

**Comment by lazebnyi** [comment]
regression tests - https://github.com/airbytehq/airbyte/actions/runs/10390977148

**Pull request merged**
